Enabling attn_mask

arashashari opened the pull request:
Although it looks fine to me, I was not able to make use of attn_mask work correctly.
Please have a look and let me know if you can find any missing part.

ptillet commented:
Thanks. I fixed the issue and merged it. There were some indexing issues in the attention mask I believe. Also added option to chose between additive or multiplicative mask for key-padding and attention mask.

arashashari commented:
Great, I will pull new changes later in the morning.
Will do a quick test and ping you in case of any questions.

Have a nice day,

Arash

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Phillippe Tillet <notifications@github.com>
Sent: Sunday, April 12, 2020 8:35:13 AM
To: ptillet/torch-blocksparse <torch-blocksparse@noreply.github.com>
Cc: Arash Ashari <Arash.Ashari@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [ptillet/torch-blocksparse] Enabling attn_mask (#3)


Thanks. I fixed the issue and merged it. There were some indexing issues in the attention mask I believe. Also added option to chose between additive or multiplicative mask for key-padding and attention mask.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fptillet%2Ftorch-blocksparse%2Fpull%2F3%23issuecomment-612634324&data=02%7C01%7CArash.Ashari%40microsoft.com%7C0098e0d90b394dfcb2a008d7def71807%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637223025157457174&sdata=EnmItjhX%2FGWZ70%2Fu1mZbvdSG%2FjZVzibpeMTkrOjdaig%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FADBJX4OKEW6AGTNYIYZO5SLRMHNTDANCNFSM4MF6OT6Q&data=02%7C01%7CArash.Ashari%40microsoft.com%7C0098e0d90b394dfcb2a008d7def71807%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637223025157467164&sdata=ljpZs0Z1LhaaXyDwy5X%2FK5Y32hk8ezNjnQwHs%2BzfuBo%3D&reserved=0>.
